### PR TITLE
Add Random.uuid_v7 to RBIs

### PR DESCRIPTION
**Pull request by dmerand**
Ruby [added support](https://github.com/ruby/ruby/commit/dfb2b4cbc9aa5edc315e210bf6bfd92fcf6e45de) for `uuid_v7` to `Random::Formatter`. This PR adds a type signature to the `core` RBIs. It also adds `Random.uuid_v4` which is an alias for `Random.uuid`.

I've set the type of `extra_timestamp_bits` to be `Integer` because that's the intent of the method. Technically anything that can be cast as `Integer` would work -- happy to adjust the type if there are differing opinions about what type to use in that situation.

### CODE REVIEW

**Review by jez — APPROVED** [review]

**Comment by dmerand** [comment]
Oh no! Committed a typo! I'll push a change.

**Comment by ilyailya** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_QFFBZfhLeS0l3Q
→ https://go/builds/bui_QFFBZL6mAYmb4z
→ https://go/builds/bui_QFFBHDXRvoSjAx

**Review by ilyailya — APPROVED** [review]

File: rbi/core/random.rbi [hunk 1]
```diff
@@ -747,4 +747,82 @@ module SecureRandom
...
+  # across distributed nodes.
+  #
+  # Counters and other mechanisms for stronger guarantees of monotonicity are
+  # not implemented.  Applications with stricter requirements should follow
+  # {Section 6.2}[https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-07.html#monotonicity_counters]
+  # of the specification.
+  sig do
+    params(extra_timestamp_bits: Integer).returns(String)
+  end
+  def self.uuid_v7(extra_timestamp_bits:); end
```
Comment (thomasmarshall) [inline rbi/core/random.rbi:827]: It looks like the default value is `0`. If this is an optional argument should it be `T.nilable(Integer)`? https://ruby-doc.org/3.3.2/stdlibs/random/Random/Formatter.html#method-i-uuid_v7

Comment (dmerand) [reply rbi/core/random.rbi:827]: Good catch! Since this has been merged, perhaps a follow-up PR would be best?

Comment (thomasmarshall) [reply rbi/core/random.rbi:827]: Sure, sounds good to me!

Comment (dmerand) [reply rbi/core/random.rbi:827]: PR created: https://github.com/sorbet/sorbet/pull/7960

**Pull request merged**